db/migrate/20181018165405_id_to_uuid_fk.rb

```diff
@@ -0,0 +1,17 @@
+require 'webdack/uuid_migration/helpers'
+
+class IdToUuidFk < ActiveRecord::Migration[5.2]
+  def change
+    reversible do |dir|
+      dir.up do
+        primary_key_and_all_references_to_uuid :applicants
+        primary_key_and_all_references_to_uuid :legal_aid_applications
+        primary_key_and_all_references_to_uuid :proceeding_types
+      end
+
+      dir.down do
+        raise ActiveRecord::IrreversibleMigration
+      end
+    end
+  end
```

Comment:
Is there any reason why you didn't just use[ `up` and `down` methods](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-up-down-methods) instead of `change` to get differing up and down behaviours:
```ruby
class IdToUuidFk < ActiveRecord::Migration[5.2]
  def up
    primary_key_and_all_references_to_uuid :applicants
    primary_key_and_all_references_to_uuid :legal_aid_applications
    primary_key_and_all_references_to_uuid :proceeding_types
  end

  def down
    raise ActiveRecord::IrreversibleMigration
  end
end
```
The result is a lot clearer I would suggest.

Reply:
I thought that because down was not actually possible (throws an error) it didn't make sense to use it in this case. However if it is best practice to use up/down even when you cannot run down to change the db I am happy to change it.

Reply:
👍

Reply:
`reversible` is useful if you only need that functionality for part of the migration. Using it for the whole migration isn't wrong - it just noisy.

Reply:
yes I can see how much clearer it looks your way